guides/advanced/building-sytems.md

```diff
@@ -277,34 +289,70 @@ Customizing your Nerves system is an advanced but powerful way to tailor the sys
 
 Customizing the build allows you to tailor the Nerves system to meet specific requirements for your hardware or application. This involves modifying Buildroot configurations and applying changes to the Nerves system.
 
-### Modify Buildroot Configuration
+### Modify Buildroot Package Configuration
 
-Nerves systems use Buildroot for building firmware. The `make menuconfig` command opens a menu-based interface where you can modify the Buildroot configuration:
+Navigate to the output directory of system you wish to modify.
 
 ```bash
 cd o/<system short name>
-make menuconfig
 ```
 
-In this interface, you can:
+Workflow for customizing a Nerves system is the same as when using Buildroot outside of Nerves,
+using `make menuconfig` and `make savedefconfig`.
 
-- Add or remove packages.
-- Configure kernel options.
-- Set custom build flags.
+The main package configuration workflows are divided into three categories,
+depending on what you want to configure:
 
-> #### Tip {: .tip}
->
-> Only make changes you understand, as incorrect settings may cause build failures or unstable firmware. For more details on Buildroot configuration, refer to the [Buildroot user manual](https://buildroot.org/downloads/manual/manual.html).
+1. Select base packages by running `make menuconfig`
+2. Modify the Linux kernel and kernel modules with `make linux-menuconfig`
+3. Enable more command line utilities using `make busybox-menuconfig`
 
-### Save the Updated Configuration
+When you quit from the `menuconfig` interface, the changes are stored
+temporarily. To save them back to your system source directory, follow the
+appropriate steps below:
 
-After making changes in `menuconfig`, save the configuration back to the system’s default configuration file (`nerves_defconfig`) using:
+1. After `make menuconfig`:
 
-```bash
-make savedefconfig
-```
+    Run `make savedefconfig` to update the `nerves_defconfig` in your System.
+
+2. After `make linux-menuconfig`:
+
+    Once done with configuring the kernel, you can save the Linux config to the
+    default configuration file using `make linux-update-defconfig`. The destination
+    file is `linux-4.9.defconfig` in your project's root (or whatever the kernel
+    version is you're working with).
+
+    > NOTE: If your system doesn't contain a custom Linux configuration yet,
+    you'll need to update the Buildroot configuration (using `make menuconfig`)
+    to point to the new Linux defconfig in your system directory. The path is
+    usually something like `$(NERVES_DEFCONFIG_DIR)/linux-x.y_defconfig`.
+
+3. After `make busybox-menuconfig`:
+
+    Unfortunately, there's not currently an easy way to save a BusyBox defconfig.
+    What you have to do instead is save the full BusyBox config and configure it
+    to be included in your `nerves_defconfig`.
+
+    ```bash
+    cp build/busybox-1.27.2/.config ../src/<full system name>/busybox_defconfig
```

Comment:
Using `<full system name>` here is not ideal.

Likely scenario is that you would want to add new system to `config.exs`
run
```bash
mix ns.clone
# Then actually just populate o directory with something like
mix ns.build --configure
# where it would just create o directory with config without building it
```

Ideally that would provide us with shell script that could export `NERVES_DEFCONFIG_DIR`
then this could be.
```bash
cp build/busybox-1.27.2/.config ${NERVES_DEFCONFIG_DIR}/busybox_defconfig
```

Reply:
It would also be good idea for build to accept `--only`
```bash
mix ns.build --only custom_rpi3
```

Reply:
So I tried this with rpi3 system and
```bash
make busybox-update-config
```
results in
```
cp -f /home/damir/Projects/nerves/nerves_systems/o/rpi3/build/busybox-1.36.1/.config /home/damir/Projects/nerves/nerves_systems/src/nerves_system_br/board/nerves-common/busybox.config
```

Which now applies it to all other systems which is IMO not wanted behavior.

In case when fragments are set it does error
```
Unable to perform busybox-update-config when fragment files are set
make[1]: *** [package/busybox/busybox.mk:464: busybox-update-config] Error 1
make: *** [Makefile:23: _all] Error 2
```

How this guide states it in current version would be correct for using fragments.

I guess my question is what would be correct way to automate
```bash
cp build/busybox-1.27.2/.config ../src/<full system name>/busybox_defconfig
```

Maybe add new make option `make busybox-update-fragment-config that would basically call
```bash
cp build/busybox-1.27.2/.config ${NERVES_DEFCONFIG_DIR}/busybox_defconfig
```
and maybe set fragments at the same time.

Reply:
Oh, you're right, if you have never modified the `busybox.config` then it goes to the default one. 

I'd do this:

```
cp build/busybox-1.27.2/.config ../src/<full system name>/busybox.config
```

Then run `make menuconfig` to set `BR2_PACKAGE_BUSYBOX_CONFIG` to `${NERVES_DEFCONFIG_DIR}/busybox.config`, and then `make savedefconfig`.

Then when I wanted to update it in the future, `make busybox-update-config` will work.

Side note: Busybox only knows how to save `.config` files. Linux and Buildroot can save `.config` and `_defconfig` versions where the `defconfig` version only has the nondefault settings.